grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -152,14 +184,16 @@ def _CreateDatapointsForTarget(
     record_values = [record.mean_resident_memory_mib for record in records_list]
   elif target == "max_resident_memory_mib":
```

Comment:
`target` comes from the HTTP request and is not validated to be exclusively mean_user_cpu_rate, max_user_cpu_rate, or max_resident_memory_mib. When provided any other value ,this function would fail in a cryptic way (because `record_values` is not initialized). Please add an else branch to throw a better exception.

Reply:
I agree; I added an `else` branch with a `NameError` exception, and also added a specific test case for it.

Just to have it written here a side note; the reason we chose to have it as a large if-else branch and not using `getattr` was to not use metaprogramming, since we know that the available resource usage metrics are rather fixed and not likely to change often.